### PR TITLE
Check-In and Notification

### PR DESCRIPTION
**Pull request by dev-zaid**
# Description
- Delete user email from check-in channel after check-in.
- Update database with user details after successful check-in.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Discord

# Screenshot
image
image

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/utils/constants.ts [hunk 1]
```diff
@@ -152,6 +158,19 @@ export const INFO = {
       message: `**Event Slug:** ${event.slug}\n**Event Name:** ${event.name}\n**Channel Started:** <#${event.checkin?.channelId}>`,
     };
   },
+  USER_CHECKIN: (teamName?: string | null) => {
```
Comment (shawshankkumar) [inline src/utils/constants.ts:161]: Why is teamName optional? And if it has to be, you need not add null.

**Review by shawshankkumar — COMMENTED** [review]
@dev-zaid Where is the code to store email id and discord id?

**Comment by dev-zaid** [comment]
> @dev-zaid Where is the code to store email id and discord id?

Line 128

File: src/service/checkin-service.ts [hunk 2]
```diff
@@ -1,268 +1,273 @@
...
+          await giveCheckinRole(incomingMessage, event.checkin!.roleId);
         }
-    } catch (err) {
-        await incomingMessage.react(CONSTANTS.checkinReactions.error);
-        serverLogger("error", incomingMessage.content, err);
+        if (event.checkin?.teamEvent) {
+          await findAndJoinTeams(incomingMessage, registrant, event);
+        }
+        await incomingMessage.delete();
+        await dbCheckin.insertOne(<checkInDBSchema>{
```
Comment (dev-zaid) [inline src/service/checkin-service.ts:128]: @shawshankkumar storing email and discord id here

File: src/utils/constants.ts (hunk 1 above)
Comment (dev-zaid) [reply src/utils/constants.ts:161]: registrantSchema has teamName Optional. That is why I have added optional here.

Comment (shawshankkumar) [reply src/utils/constants.ts]: Okay so all you need to do is write teanName?:string and it automatically understands that.

**Review by shawshankkumar — APPROVED** [review]
LGTM

**Pull request merged**